rpcs3/Emu/Cell/lv2/sys_spu.cpp

```diff
@@ -159,21 +160,39 @@ error_code _sys_spu_image_get_information(vm::ptr<sys_spu_image> img, vm::ptr<u3
 	return CELL_OK;
 }
 
+static error_code spu_image_open(vm::ptr<sys_spu_image> img, fs::file file)
+{
+	img->load(decrypt_self(std::move(file), fxm::get_always<LoadedNpdrmKeys_t>()->devKlic.data()));
+
+	return CELL_OK;
+}
+
 error_code sys_spu_image_open(vm::ptr<sys_spu_image> img, vm::cptr<char> path)
 {
 	sys_spu.warning("sys_spu_image_open(img=*0x%x, path=%s)", img, path);
 
-	const fs::file elf_file = decrypt_self(fs::file(vfs::get(path.get_ptr())), fxm::get_always<LoadedNpdrmKeys_t>()->devKlic.data());
+	fs::file file = fs::file(vfs::get(path.get_ptr()));
 
-	if (!elf_file)
+	if (!file)
 	{
-		sys_spu.error("sys_spu_image_open() error: failed to open %s!", path);
 		return CELL_ENOENT;
 	}
 
-	img->load(elf_file);
+	return spu_image_open(img, std::move(file));
```

Comment:
I'm only using std::move for it to compile due to deleted constructors, the object isn't being used later so I think it's fine.